_source/_posts/2019-09-24-how-to-create-a-symfony-app-with-authentication.md

```diff
@@ -0,0 +1,771 @@
+
+[Register for a forever-free developer account](https://developer.okta.com/signup/), then log in to your Okta account, go to the Applications menu item and click the **Add Application** button:
+
+{% img blog/symfony-authentication/add-application.png alt:"Add Application button" width:"260" %}{: .center-image }
+  
+Select **Web** and click **Next**.
+
+{% img blog/symfony-authentication/choose-web-app.png alt:"Select the application type" width:"800" %}{: .center-image }
+
+Enter a title, and set `http://localhost:8000/` as the Base URI, and `http://localhost:8000/authorization-code/callback` as the Login Redirect URI, then click **Done**. You can leave the rest of the settings as they are.
+
+Copy the Client ID and Client Secret from the application settings, you will need them soon.
+
+## Set up the Symfony Project
+
+Create a MySQL database and user:
+
+```bash
+mysql -uroot -p
+CREATE DATABASE symfony_auth CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
+GRANT ALL on symfony_auth.* to 'symfony_auth'@'127.0.0.1' identified by 'symfony_auth';
+quit
+```
+
+Initialize a new Symfony project using the Flex skeleton and start the built-in PHP server:
+
+```bash
+composer create-project symfony/skeleton symfony-auth-demo
+cd symfony-auth-demo
+php -S 127.0.0.1:8000 -t public
+```
+
+Load `http://127.0.0.1:8000` and you should see the default Symfony page.
+
+Make sure the version of Symfony is locked to 4.3 in `composer.json` by confirming the versions match the below:
+
+```
+...
+        "symfony/console": "4.3.*",
+        "symfony/dotenv": "4.3.*",
+        "symfony/flex": "^1.3.1",
+        "symfony/framework-bundle": "4.3.*",
+        "symfony/yaml": "4.3.*"
+...
+```
+
+Update the dependencies if you made any changes to `composer.json`:
+
+```bash
+composer update
+```
+
+Create a new `.env.local` file inside the main project directory to store your local credentials and add the following values:
+
+`.env.local`
+
+```
+DATABASE_URL=mysql://symfony_auth:symfony_auth@127.0.0.1:3306/symfony_auth
+OKTA_BASE_URL=http://localhost:8000/
+OKTA_REDIRECT_URI=http://localhost:8000/authorization-code/callback
+OKTA_CLIENT_ID=
+OKTA_CLIENT_SECRET=
+OKTA_METADATA_URL=https://{{yourOktaDomain}}/oauth2/default/.well-known/oauth-authorization-server
+```
+
+Replace the domain name in `OKTA_METADATA_URL` with your own (`dev-xxxxxx.okta.com`), and input your Client ID and Client Secret (saved when you created the Okta application).
+
+## Create a Basic Symfony Controller and Home Page Template
+
+In this section, you'll create your first Symfony controller and a simple home page template. Since you started with Symfony Flex and just the skeleton, add the Framework Extra bundle (so you can use the Doctrine ORM and Annotations) and the templating engine (Twig) bundle:
+
+```bash
+composer require sensio/framework-extra-bundle symfony/twig-bundle
+```
+
+Replace `templates/base.html.twig`:
+
+{%raw%}
```

Comment:
Raw tags are only needed for double curlies: {{ }}

Reply:
That's not what I saw. Without the raw tag I get this error here:

> Liquid Exception: Liquid syntax error (line 115): Unknown tag 'block'